# THA3: Research question and dataset.
*  Find a simple research question and compile a dataset to answer that question.
* The dataset must be compiled from various sources.
*  Compile a data dictionary for your dataset.
*  Update your GitHub repository accordingly.

## Research Question

I want to study the relationship between foundations' evaluation result and their donation income. Ideally, organization's evaluation result is a kind of important information for public and donors to make their donation decision. Evaluation result should help the nonprofit sector develop the mechanism of selecting the superior organizaitons and eliminating the inferior ones. However, empirical studies on this topic are limit, and researches has fallen short of analysing what factors could contribute to the relationship. My research questions are as follow:
* Whether the evaluation result and foundation's donation income have correlationship? 

## Data Source: RICF
I will use 2016 annual report's data in RICF, which is the lastest data of China's foundation. In order to answer my question, I will merge two dataset as following:

Basic information: https://github.com/ma-ji/RICF/raw/master/2016/2016_basic.tsv

Financial_activities: https://raw.githubusercontent.com/ma-ji/RICF/master/2016/2016_financial_activities.tsv

Codebooks: https://github.com/ma-ji/RICF/blob/master/RICF%20Codebook.xlsx




## Research Method and Workflow



### 1. Read and extract evaluation result  from 2016 basic information

In [7]:
import pandas as pd
df1= pd.read_csv("https://raw.githubusercontent.com/ma-ji/RICF/master/2016/2016_basic.tsv",sep='\t')
df_bas=df1[["ricf_oid","ba_evi"]]  
df_bas.rename(columns={'ba_evi':'ba_evl'}, inplace = True)
print(df_bas.head(5))

   ricf_oid ba_evl
0      3490     1A
1      5690     --
2      3070    未参评
3      1583    未参评
4      3884     --


/usr/local/lib/python3.6/dist-packages/pandas/core/frame.py:4025: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  return super(DataFrame, self).rename(**kwargs)


### 2. Read and extract donation income from financial activities

In [8]:
df2= pd.read_csv('https://raw.githubusercontent.com/ma-ji/RICF/master/2016/2016_financial_activities.tsv',sep='\t')
list(df2)
df_act=df2[['ricf_oid','fa_dine_this','fa_dinu_this']]
print(df_act.head(5))


   ricf_oid fa_dine_this  fa_dinu_this
0      2161         0.00    100,000.00
1      1364         0.00  3,272,400.40
2      2213         0.00          0.00
3      3889         0.00  3,073,774.50
4      5095         0.00  2,016,000.00


### 3. Meger two datasets

In [9]:
df_3=df_bas.merge(df_act, on="ricf_oid", how="inner")
print(df_3.head(5))


   ricf_oid ba_evl   fa_dine_this  fa_dinu_this
0      2288    未参评           0.00    379,538.00
1      2256    未参评   1,273,789.30  4,745,780.90
2      1059    未参评           0.00      9,200.00
3       379     1A  16,793,400.00          0.00
4       972    未参评           0.00    709,826.90


In [0]:
df_3.ba_evl.fillna('None').str.contains('A')
df_e=df_3[df_3.ba_evl.fillna('None').str.contains('A')]
df_e['ba_evl']=[s.strip('A') for s in df_e.ba_evl]
df_e


In [12]:
df_e.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 511 entries, 3 to 1749
Data columns (total 4 columns):
ricf_oid        511 non-null int64
ba_evl          511 non-null object
fa_dine_this    511 non-null object
fa_dinu_this    511 non-null object
dtypes: int64(1), object(3)
memory usage: 20.0+ KB


### 4. Clean data

In [0]:
df_e['ba_evl'] = df_e['ba_evl'].astype('float64')
df_e

In [14]:
df_e['fa_dine_this']=df_e['fa_dine_this'].apply(lambda x: x.replace(',', '')).astype('float')
df_e['fa_dinu_this']=df_e['fa_dinu_this'].apply(lambda x: x.replace(',', '')).astype('float')

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [0]:
# doi= donation income of 2016
df_e["doi"]=df_e["fa_dine_this"]+df_e["fa_dinu_this"]
df_e

### 5. Correlation test

In [0]:
df_cor=df_e[['doi','ba_evl']]
df_cor

In [0]:
df_cor_s=df_cor.corr(method='spearman')
df_cor_s

doi    ba_evl
doi     1.000000  0.317896
ba_evl  0.317896  1.000000

###6. Conclution

 The evaluation result and donation income have correlationship, the correlation coefficient R= 0.317896

